### ISSUE TITLE
After upgrading from 0.97 to .100.3, it seems impossible to read a cell's value without knowing its type first

### ISSUE THREAD

**Issue by sjd2021** [ClosedXML/ClosedXML#2054]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [ ] Bug
- [ ] Feature
- [x] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [x] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

e.g. 0.95.3

**What is the current behavior?**

Simply trying to get cell.Value throws a VerificationException 

**What is the expected behavior or new feature?**

Get a value of any type given a cell 

**Is this a regression from the previous version?**

Yes

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
   var stream = File.OpenRead("pathhere");
   var workbook = new XLWorkbook(stream);
   var cell = workbook.Worksheets.First().Cell(1);

   var textValue = cell.GetValue<string>(); // successfully pulls string value from text field
   cell.TryGetValue<dynamic>(out var dynamicValue); // always null.. useless
   var rawValue = cell.Value; // always throws VerificationException
}
```

[Sample.xlsx](https://github.com/ClosedXML/ClosedXML/files/11182663/Sample.xlsx)

- [ x ] I attached a sample spreadsheet.  (You can drag files on to this issue)

Previously, `dynamicValue` worked fine. I saw that there's been a change such that cell values aren't just implicitly converted between strings/objects all the time anymore, but I'm not sure how I can read the value of any cell without knowing its type. I'll gladly convert everything to a string if that somehow works, but that doesn't seem right. Why I can't read `cell.Value` at all is the weirdest part.

**Comment by sjd2021** [comment]
I guess I'm just struggling to figure out how we can reference the raw value without doing a whole bunch of "if this, GetValue<x>" checks.

**Comment by jahav** [comment]
@sjd2021 If possible, use `IXLCell.Value` property which now returns `XLCellValue` structure, unlike pre-0.100 type `object`. I kept `GetValue`, but mostly for compatibility reasons. It should mostly work the same, but change in semantic was significant.

For more info, please see [release notes](https://github.com/ClosedXML/ClosedXML/releases/tag/0.100.0) and [migration guide](https://closedxml.readthedocs.io/en/latest/migrations/migrate-to-0.100.html#migration-from-0-97-to-0-100).

**Comment by jahav** [comment]
The provided sample doesn't throw exceptions. It needed a proper cell address `workbook.Worksheets.First().Cell(1, 1)` to compile, but that's all.

**Comment by D9001** [comment]
@jahav I figured it out. It failed while using the debugger consistently. That's where I was typing all this. I'm guessing the verification exception has something to do with two threads(?) attempting to read the cell or something..

**Comment by sjd2021** [comment]
@jahav  That was just an example of my poor copy/paste skills.  I figured it out. It failed while using the debugger consistently. That's where I was typing all this. I'm guessing the verification exception has something to do with two threads(?) attempting to read the cell or something that only happens in debug.

In order to get it as an object, I just ended up making a switch to call GetXyz based on the data type. Not having this prevents you from performing generic mapping operations that previously worked since the underlying type didn't need to be known in order to access its value. I understand the reasoning for the change though.
